metaflow/plugins/datatools/s3/s3.py

```diff
@@ -792,6 +792,59 @@ def _head():
 
         return list(starmap(S3Object, _head()))
 
+    def delete(self, key):
+        """
+        Delete a single object in S3.
+
+        Parameters
+        ----------
+        key : str
+            Object to delete. It can be an S3 url or a path suffix.
+        """
+
+        url = self._url(key)
+        src = urlparse(url)
+
+        def _delete(s3, tmp):
+            resp = s3.delete_object(Bucket=src.netloc, Key=src.path.lstrip('/'))
```

Comment:
can simplify to be `return s3.delete_object(...)`

Reply:
also need to protect against key = None; it should be legit (and to follow the other calls, we can make key optional) and corresponds to deleting the root.